sktime/forecasting/model_selection/_tune.py

```diff
@@ -719,3 +720,250 @@ def get_test_params(cls, parameter_set="default"):
         }
 
         return [params, params2]
+
+
+class ForecastingSkoptSearchCV(BaseGridSearch):
+    """Bayesian search over hyper parameters for a forecaster.
+
+    skopt version 0.9.0 (under-development) - 90% done
+
+    Parameters
+    ----------
+    search_spaces : dict or a list of dic. for each dict, keys are parameters names
+    (string) and values are skopt space Real, Interger, Object.
+    """
+
+    def __init__(
+        self,
+        forecaster,
+        cv,
+        param_grid,
+        n_iter=10,
+        n_points=1,
+        random_state=None,
+        scoring=None,
+        strategy="refit",
+        n_jobs=None,
+        refit=True,
+        verbose=0,
+        return_n_best_forecasters=1,
+        pre_dispatch="2*n_jobs",
+        backend="loky",
+        update_behaviour="full_refit",
+        error_score=np.nan,
+        optimizer_kwargs=None,
+    ):
+        _check_soft_dependencies(
```

Comment:
non-blocking side note: I suppose we should think whether we can cover this with a minor update to the tag system?
We have three instances now, `scikit-learn`/`sklearn`, `dtw-python`/`dtw_python`, and this

Reply:
I'm not really familiar with the tag system yet. could you point me to where to look to understand how this tagging system?. Also, to verify, did I use the `_check_soft_dependecies` and import the dependency in the class correctly?

Reply:
oh yes, you did it correctly!
Sorry if I wasn't clear, well done inferring the pattern from the few examples (or guessing) where import path does not equal the package name.

What I meant, maybe you have a suggestion how this could be improved for developers as it's a bit clunky and an odd special case.

Regarding general information on the tag system, it's not in the developer guide, but the best references are:

* the extension templates, they contain a full documentation of the tags and what they do
* as a tutorial on "writing your own estimator", the 2 hour workshop here: https://github.com/sktime/sktime-workshop-pydata-london-2022 (on video), which also contains explanations of the tag system and the most important tags

Reply:
I solved the issues with softdeps tests by including a `python_dependecies` tag to the class but I'm still would like to know what kind of test the `get_test_params` does.

Reply:
resolved here #4832 and #4867